src/libexpr/primops.cc

```diff
@@ -3942,6 +4072,13 @@ static RegisterPrimOp primop_genList({
       ```
 
       returns the list `[ 0 1 4 9 16 ]`.
+
+      # Time Complexity
+
+      O(n * T_f) where:
+
+      n = requested length
+      T_f = generator function evaluation time
```

Comment:
```suggestion
      O(n) where:

      n = requested length
      
      Calls to `generator` are performed afterwards when needed.
```

Reply:
Idea from a call with @roberth, use both notations

Comlexity of `genList n f`: O(n) 

Comlexity of `deepSeq (genList n f)`: O(n * T_f) 

where

T_f = generator `f` function evaluation time